static/js/components/AnalysisServicePage.jsx

```diff
@@ -0,0 +1,205 @@
+}
+
+export function analysisServiceInfo(analysisService) {
+  if (!analysisService?.url) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const share_groups = [];
+  analysisService.groups.forEach((share_group) => {
+    share_groups.push(share_group.name);
+  });
+
+  let result = `Description: ${analysisService.description} / URL: ${analysisService.url}`;
+
+  if (share_groups.length > 0) {
+    result += "\r\n(";
+    result += `Default Share Groups: ${share_groups.join(", ")}`;
+    result += ")";
+  }
+
+  return result;
+}
+
+const AnalysisServiceList = ({ analysisServices, deletePermission }) => {
+  const dispatch = useDispatch();
+  const classes = useStyles();
+  const textClasses = textStyles();
+  const groups = useSelector((state) => state.groups.all);
+
+  const deleteAnalysisService = (analysisService) => {
+    dispatch(
+      analysisServiceActions.deleteAnalysisService(analysisService.id)
+    ).then((result) => {
+      if (result.status === "success") {
+        dispatch(showNotification("AnalysisService deleted"));
+      }
+    });
+  };
+
+  return (
+    <div className={classes.root}>
+      <List component="nav">
+        {analysisServices?.map((analysisService) => (
+          <ListItem button key={analysisService.id}>
+            <ListItemText
+              primary={analysisServiceTitle(analysisService)}
+              secondary={analysisServiceInfo(analysisService, groups)}
+              classes={textClasses}
+            />
+            <Button
+              key={analysisService.id}
+              id="delete_button"
+              classes={{
+                root: classes.analysisServiceDelete,
+                disabled: classes.analysisServiceDeleteDisabled,
+              }}
+              onClick={() => deleteAnalysisService(analysisService)}
+              disabled={!deletePermission}
+            >
+              &times;
+            </Button>
+          </ListItem>
+        ))}
+      </List>
+    </div>
+  );
+};
+
+const AnalysisServicePage = () => {
+  const { analysisServiceList } = useSelector(
+    (state) => state.analysis_services
+  );
+
+  const currentUser = useSelector((state) => state.profile);
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  const permission =
+    currentUser.permissions?.includes("System admin") ||
+    currentUser.permissions?.includes("Manage Analysis Services");
+
+  useEffect(() => {
+    const getAnalysisServices = async () => {
+      // Wait for the allocations to update before setting
+      // the new default form fields, so that the instruments list can
+      // update
+
+      await dispatch(analysisServicesActions.fetchAnalysisServices());
+    };
+
+    getAnalysisServices();
+
+    // Don't want to reset everytime the component rerenders and
+    // the defaultStartDate is updated, so ignore ESLint here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is it weird that I pay more attention to the code comments than the code when I'm reviewing?

Reply:
removing :)